internal/archive/archive.go

```diff
@@ -143,7 +143,7 @@ func ReadTarEntry(rc io.Reader, entryPath string) (*tar.Header, []byte, error) {
 			return nil, nil, errors.Wrap(err, "failed to get next tar entry")
 		}
 
-		if header.Name == entryPath {
+		if filepath.Clean(header.Name) == entryPath {
```

Comment:
Do we want to clean both sides?

Reply:
I was imagining that since we are the only users of this function we can always pass in paths w/o `./`, whereas we should be able to handle any reasonable tar we get.